src/View/Helper/FormHelper.php

```diff
@@ -2407,7 +2407,11 @@ protected function _initInputField($field, $options = [])
             unset($options['value']);
         }
         if (!isset($options['val'])) {
-            $options['val'] = $context->val($field);
+            $valOptions = [
+                'default' => isset($options['default']) ? $options['default'] : null,
+                'schemaDefault' => isset($options['schemaDefault']) ? $options['schemaDefault'] : true,
+            ];
+            $options['val'] = $context->val($field, $valOptions);
```

Comment:
Yup, I will. Was waiting for green signal on the idea in general before putting in more effort :)